src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const keyValue = item[key];
+
+    if (!((keyValue as unknown as string) in result)) {
+      result[keyValue as unknown as string] = [];
```

Comment:
Using type assertions to cast `keyValue` as `unknown` and then as `string` is unsafe and can lead to runtime errors if the key value is not a string. Instead, ensure that `key` is always a property with string values, or handle the case where `keyValue` is not a string.